pnet_sys/src/windows.rs

```diff
@@ -1,112 +1,224 @@
-    pub const IFF_POINTTOPOINT: libc::c_int = 0x00000008;
-    pub const IFF_POINTOPOINT: libc::c_int = IFF_POINTTOPOINT;
-    pub const IFF_MULTICAST: libc::c_int = 0x00000010;
-
-    pub const INVALID_SOCKET: CSocket = winapi::INVALID_SOCKET;
 
+    use std::io;
+    use std::mem;
+    use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, SocketAddrV6};
+    use super::winapi::ctypes;
+    use super::winapi::shared::{inaddr, in6addr, ws2def, ws2ipdef};
+    use super::winapi::um::winsock2;
+    use super::{htons, ntohs};
+
+    pub type CSocket = winsock2::SOCKET;
+    pub type Buf = *const ctypes::c_char;
+    pub type MutBuf = *mut ctypes::c_char;
+    pub type BufLen = ctypes::c_int;
+    pub type CouldFail = ctypes::c_int;
+    pub type SockLen = ctypes::c_int;
+    pub type SockAddr = ws2def::SOCKADDR;
+    pub type SockAddrIn = ws2def::SOCKADDR_IN;
+    pub type SockAddrIn6 = ws2ipdef::SOCKADDR_IN6;
+    pub type SockAddrStorage = ws2def::SOCKADDR_STORAGE;
+    pub type SockAddrFamily = ws2def::ADDRESS_FAMILY;
+    pub type SockAddrFamily6 = ws2def::ADDRESS_FAMILY;
+    pub type InAddr = inaddr::IN_ADDR;
+    pub type In6Addr = in6addr::IN6_ADDR;
+
+    pub const AF_INET: ctypes::c_int = ws2def::AF_INET;
+    pub const AF_INET6: ctypes::c_int = ws2def::AF_INET6;
+    pub const SOCK_RAW: ctypes::c_int = winsock2::SOCK_RAW;
+
+    pub const IPPROTO_IP: ctypes::c_int = ws2def::IPPROTO_IP;
+    pub const IP_HDRINCL: ctypes::c_int = ws2ipdef::IP_HDRINCL;
+    pub const IP_TTL: ctypes::c_int = ws2ipdef::IP_TTL;
+
+    pub const IFF_UP: ctypes::c_int = 0x00000001;
+    pub const IFF_BROADCAST: ctypes::c_int = 0x00000002;
+    pub const IFF_LOOPBACK: ctypes::c_int = 0x00000004;
+    pub const IFF_POINTTOPOINT: ctypes::c_int = 0x00000008;
+    pub const IFF_POINTOPOINT: ctypes::c_int = IFF_POINTTOPOINT;
+    pub const IFF_MULTICAST: ctypes::c_int = 0x00000010;
+
+    pub const INVALID_SOCKET: CSocket = winsock2::INVALID_SOCKET;
 
     pub unsafe fn close(sock: CSocket) {
-        let _ = ws2_32::closesocket(sock);
+        let _ = winsock2::closesocket(sock);
     }
 
-    pub unsafe fn socket(af: libc::c_int, sock: libc::c_int, proto: libc::c_int) -> CSocket {
-        ws2_32::socket(af, sock, proto)
+    pub unsafe fn socket(af: ctypes::c_int, sock: ctypes::c_int, proto: ctypes::c_int) -> CSocket {
+        winsock2::socket(af, sock, proto)
     }
 
     pub unsafe fn setsockopt(socket: CSocket,
-                            level: libc::c_int,
-                            name: libc::c_int,
+                            level: ctypes::c_int,
+                            name: ctypes::c_int,
                             value: Buf,
                             option_len: SockLen)
-        -> libc::c_int {
-        ws2_32::setsockopt(socket, level, name, value, option_len)
+        -> ctypes::c_int {
+        winsock2::setsockopt(socket, level, name, value, option_len)
+    }
+
+    pub fn addr_to_sockaddr(addr: SocketAddr,
```

Comment:
Hmm - were these duplicated when you moved them? Is there a reason to have them multiple times rather than just once?

Reply:
There was a reason, can't say for certain it was a "good" reason. See above comment reply.